zomes/holo_acorn/code/src/lib.rs

```diff
@@ -244,8 +252,21 @@ mod holo_acorn {
             validation_package: || {
                 hdk::ValidationPackageDefinition::Entry
             },
-            validation: | _validation_data: hdk::EntryValidationData<Goal>| {
-                Ok(())
+            validation: | validation_data: hdk::EntryValidationData<Goal>| {
+                match validation_data {
+                    EntryValidationData::Modify {
+                        mut new_entry,
+                        old_entry,
+                        ..
+                     }=>{
+                         if new_entry.timestamp_updated==None
+                         {new_entry.timestamp_updated=Some(new_entry.timestamp_created);}
+                         new_entry.timestamp_created=old_entry.timestamp_created;
+                         new_entry.user_edit_hash=Some(AGENT_ADDRESS.clone());
```

Comment:
what does this code do?

Reply:
that code makes sure that timestamp_created remains always the same, the new incoming timestamp_created becomes the timestamp_update to avoid having to change anything in the interface other than the name the attribute and it also saves the hash of the user who performed the update

Reply:
aha. 
Actually, validation code cannot MODIFY the entry. Those checks should be performed in the main `update_goal` call actually. 

An important concept in Holochain: the validation code will be run by ANY NODE who is receiving the entry, via p2p gossip.

Reply:
It's okay. I'll fix it.